index.js

```diff
@@ -6,41 +6,28 @@ const path = require('path')
 module.exports = function generateRequireForUserCode (scopedDirs, options) {
   options = _.defaults((options || {}), {autoDeleteCache: false})
 
-  const forExtensions = Object.keys(require.extensions)
-  const uniqueIdForThisScopedRequire = _.uniqueId('__dontExtendThisScopedRequire')
   scopedDirs = _.map(scopedDirs, function (dir) { return path.resolve(dir) })
 
   const baseModule = require('./lib/stubmodule-that-does-the-require')
   // so that it can be re-used again with another scoped-dir, I delete it from the cache
   delete Module._cache[baseModule.id]
   // make relative paths work when requiring
   baseModule.filename = path.resolve(scopedDirs[0], 'stubmodule-that-does-the-require.js')
-  baseModule.__scopedRequireModule = true
 
-  function inUserCodeDirs (modulePath) {
-    return _.some(scopedDirs, function (userCodeDir) { return modulePath.indexOf(userCodeDir) >= 0 })
+  function addPaths (m) {
+    m.paths = m.paths.concat(scopedDirs)
   }
 
-  function adjustPaths (m) {
-    m.paths = _.filter(m.paths.concat(scopedDirs), function (modulePath) { return inUserCodeDirs(modulePath) })
-  }
-
-  adjustPaths(baseModule)
+  addPaths(baseModule)
 
-  _.forEach(forExtensions, function (ext) {
-    const original = require.extensions[ext]
-    if (original && original[uniqueIdForThisScopedRequire]) { return }
+  _.forEach(require.extensions, function (extensionFunc, extension) {
+    const original = extensionFunc
```

Comment:
do we need this variable now that we have `extensionFunc` ?

also as a side note I would probably rename it to `extensionLoader`